--- 87011.user.js	2013-05-04 07:23:49.000000000 +0200
+++ 87011.user_new.js	2013-05-04 07:37:11.914055785 +0200
@@ -776,7 +776,7 @@
      
     /* My Player Window */
     var myPlayerWindow = createMyElement ('div', '', '', '', '');
-    styleMyElement (myPlayerWindow, {position: 'relative', width: '640px', height: '390px', backgroundColor: '#F4F4F4', zIndex: '99999'});
+    styleMyElement (myPlayerWindow, {position: 'relative', width: '640px', height: '416px', backgroundColor: '#F4F4F4', zIndex: '99999'});
     appendMyElement (ytPlayerWindow, myPlayerWindow);
     
     /* Get Video Thumbnail */
@@ -875,9 +875,9 @@
 	  'videoThumb': ytVideoThumb,
 	  'videoTitle': ytVideoTitle,
 	  'playerWidth': 640,
-	  'playerHeight': 390,
+	  'playerHeight': 416,
 	  'playerWideWidth': 970,
-	  'playerWideHeight': 510,
+	  'playerWideHeight': 532,
 	  'sidebarWindow': ytSidebarWindow,
 	  'sidebarMarginNormal': -390,
 	  'sidebarMarginWide': 10
@@ -912,7 +912,9 @@
   else {
     /* My Player Window */
     var myPlayerWindow = createMyElement ('div', '', '', '', '');
-    styleMyElement (myPlayerWindow, {position: 'relative', width: '620px', height: '360px', backgroundColor: '#F4F4F4', zIndex: '99999'});
+    var PlayerHeight = dmPlayerWindow.clientHeight + 22;
+    var PlayerWidth = dmPlayerWindow.clientWidth;
+    styleMyElement (myPlayerWindow, {position: 'relative',width: playerWidth + 'px', height: playerHeight + 'px', backgroundColor: '#F4F4F4', zIndex: '99999'});
     modifyMyElement (dmPlayerWindow, 'div', '', true);
     styleMyElement (dmPlayerWindow, {height: '100%'});
     appendMyElement (dmPlayerWindow, myPlayerWindow);
@@ -953,8 +955,8 @@
 	  'videoList': dmVideoList,
 	  'videoPlay': dmDefaultVideo,
 	  'videoThumb': dmVideoThumb,
-	  'playerWidth': 620,
-	  'playerHeight': 360,
+	  'playerWidth': playerWidth,
+	  'playerHeight': playerHeight,
 	  'playerWideWidth': 970,
 	  'playerWideHeight': 510,
 	  'sidebarWindow': dmSidebarWindow,
@@ -989,7 +991,9 @@
   else {
     /* My Player Window */    
     var myPlayerWindow = createMyElement ('div', '', '', '', '');
-    styleMyElement (myPlayerWindow, {position: 'relative', width: '960px', height: '540px', backgroundColor: '#F4F4F4', zIndex: '99999'});
+    var PlayerHeight = viPlayerWindow.clientHeight + 22;
+    var PlayerWidth = viPlayerWindow.clientWidth;
+    styleMyElement (myPlayerWindow, {position: 'relative', width: PlayerWidth + 'px', height: PlayerHeight + 'px', backgroundColor: '#F4F4F4', zIndex: '99999'});
     modifyMyElement (viPlayerWindow, 'div', '', true);
     styleMyElement (viPlayerWindow, {height: '100%'});
     appendMyElement (viPlayerWindow, myPlayerWindow);
@@ -1029,7 +1033,7 @@
       if (viVideoFound) {
 	/* Create Player */
 	var viDefaultVideo = 'Low Definition MP4';
-	player = {'playerSocket': viPlayerWindow, 'playerWindow': myPlayerWindow, 'videoList': viVideoList, 'videoPlay': viDefaultVideo, 'videoThumb': viVideoThumb, 'playerWidth': 960, 'playerHeight': 540};
+	player = {'playerSocket': viPlayerWindow, 'playerWindow': myPlayerWindow, 'videoList': viVideoList, 'videoPlay': viDefaultVideo, 'videoThumb': viVideoThumb, 'playerWidth': PlayerWidth, 'playerHeight': PlayerHeight};
 	feature['container'] = false;
 	feature['widesize'] = false;
 	option['definitions'] = ['High Definition', 'Low Definition', 'Very Low Definition'];
@@ -1059,7 +1063,9 @@
   else {
     /* My Player Window */
     var myPlayerWindow = createMyElement ('div', '', '', '', '');
-    styleMyElement (myPlayerWindow, {position: 'relative', width: '640px', height: '364px', backgroundColor: '#F4F4F4', zIndex: '99999'});
+    var PlayerHeight = mcPlayerWindow.clientHeight + 22;
+    var PlayerWidth = mcPlayerWindow.clientWidth;
+    styleMyElement (myPlayerWindow, {position: 'relative', width: PlayerWidth + 'px', height: PlayerHeight + 'px', backgroundColor: '#F4F4F4', zIndex: '99999'});
     modifyMyElement (mcPlayerWindow, 'div', '', true);
     styleMyElement (mcPlayerWindow, {height: '100%'});
     appendMyElement (mcPlayerWindow, myPlayerWindow);
@@ -1117,8 +1123,8 @@
 	  'videoList': mcVideoList,
 	  'videoPlay': mcDefaultVideo,
 	  'videoThumb': mcVideoThumb,
-	  'playerWidth': 640,
-	  'playerHeight': 364,
+	  'playerWidth': PlayerWidth,
+	  'playerHeight': PlayerHeight,
 	  'playerWideWidth': 960,
 	  'playerWideHeight': 510,
 	  'sidebarWindow': mcSidebarWindow,
